src/main/java/org/sharedid/endpoint/handler/CheckGdprConsentHandler.java

```diff
@@ -0,0 +1,174 @@
+package org.sharedid.endpoint.handler;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import io.vertx.core.http.HttpServerRequest;
+import org.sharedid.endpoint.consent.GdprConsentString;
+import org.sharedid.endpoint.context.DataContext;
+import org.sharedid.endpoint.service.LocationService;
+import org.sharedid.endpoint.util.GdprUtil;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.ext.web.RoutingContext;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CheckGdprConsentHandler implements Handler<RoutingContext> {
+    private static final Logger logger = LoggerFactory.getLogger(CheckGdprConsentHandler.class);
+
+    public static final int SHARED_ID_VENDOR_ID = 52;
```

Comment:
Vendor ID 52 is for Rubicon Project. I think Shared ID suppose to use a different vendor ID.

Reply:
This has come up before. @bretg, are we going to register prebid with the IAB to get a vendor ID, or will we continue with using the Magnite specific vendor for now?

Reply:
We're going to register prebid/shared id with the IAB and get an alternate vendor ID. I'll replace the above with the correct value once that submission is complete.